lsp/nls/src/requests/completion.rs

```diff
@@ -80,22 +109,37 @@ lazy_static! {
     // This regexp must be the reverse of the regexp in the lexer (nickel_lang::parser::lexer)
     // to correctly parse identifiers.
     static ref RE: regex::Regex = regex::Regex::new(r"[_a-zA-Z0-9-']*[a-zA-Z]_?").unwrap();
+    static ref RE_SPACE: regex::Regex = regex::Regex::new(r"\s+").unwrap();
 }
 
-/// Get the identifier before the `.`, for record completion.
-fn get_identifier_before_dot(text: &str) -> Option<String> {
+/// Get the string chunks that make up an identifier path.
+fn get_identifier_path(text: &str) -> Option<Vec<String>> {
+    // skip any `.` at the end of the text
     let text: String = text
         .chars()
         .rev()
         .skip_while(|c| *c == '.') // skip . (if any)
+        .collect::<String>()
```

Comment:
I assume you've added the `collect` call here because of [this issue](https://stackoverflow.com/a/69782688). However, we can avoid allocating the string twice by just cloning `text` immediately then repeatedly popping the last character of the string if it's a dot. E.g.

```rust
let text = {
    let mut text = text.to_string;
    while text.ends_with('.') {
         text.pop();
    }
    text
};
```

I've added the extra scope here to limit the mutability of the string to just that expression. I appreciate this feels a lot less FP-ish than what was here before, but I don't think it's too problematic to use a little bit of mutation here to avoid an entire string copy.

Reply:
Yes, it is because of that issue 😅 This also looks good since the scope limited.